Test/DependencyCollector/FunctionalTests/FuncTest/SqlTests.cs

```diff
@@ -372,7 +387,7 @@ private void ExecuteSyncSqlTests(TestWebApplication testWebApplication, int expe
                     //// The above request would have trigged RDD module to monitor and create RDD telemetry
                     //// Listen in the fake endpoint and see if the RDDTelemtry is captured
 
-                    var allItems = DeploymentAndValidationTools.SdkEventListener.ReceiveAllItemsDuringTimeOfType<TelemetryItem<RemoteDependencyData>>(DeploymentAndValidationTools.SleepTimeForSdkToSendEvents);
+                    var allItems = DeploymentAndValidationTools.SdkEventListener.ReceiveAllItemsDuringTimeOfType<TelemetryItem<RemoteDependencyData>>(DeploymentAndValidationTools.SleepTimeForSdkToSendEvents).ToArray();
```

Comment:
What is the value of this first ToArray()? The next line will just iterate again in the Where which should make this an expensive waste of a conversion and object creation.